batcher/aligned-batcher/src/lib.rs

```diff
@@ -285,25 +295,30 @@ impl Batcher {
         if self.user_balance_is_unlocked(addr).await {
             return false;
         }
+        let mut batch_state = self.batch_state.lock().await;
 
-        let mut user_proof_counts = self.user_proof_count_in_batch.lock().await;
-        let user_proofs_in_batch = *user_proof_counts.get(addr).unwrap_or(&0) + 1;
-
+        let user_proofs_in_batch = batch_state
+            .user_proof_count_in_batch
+            .get(addr)
+            .unwrap_or(&0)
+            + 1;
         let user_balance = self.get_user_balance(addr).await;
 
         let min_balance = U256::from(user_proofs_in_batch) * U256::from(MIN_BALANCE_PER_PROOF);
         if user_balance < min_balance {
             return false;
         }
 
-        user_proof_counts.insert(*addr, user_proofs_in_batch);
+        batch_state
+            .user_proof_count_in_batch
+            .insert(*addr, user_proofs_in_batch);
```

Comment:
let's define a method on `BatchState` with this functionality too. Also, let's do this update outside `check_user_balance`, after ir returns `true`

Reply:
Solved in [5e72f95](https://github.com/yetanotherco/aligned_layer/pull/679/commits/5e72f951b9886bef437e6f333b5c8db2b8613ec8)